src/backend/backup.py

```
@@ -1,196 +1,356 @@
+
+        Returns:
+            dict: Backup information
+        """
+        try:
+            from bson import ObjectId
+            from bson.errors import InvalidId
+
+            try:
+                object_id = ObjectId(backup_id)
+            except InvalidId:
+                raise ValueError("Invalid backup ID format")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)